src/scss/parse.js

```diff
@@ -4038,106 +4023,257 @@ function _checkUnicodeWildcard(i) {
+  const leftParenthesis = tokens[i];
+
+  // Skip (
+  i++;
+
+  // Determine the type of URI
+  if (checkUri1(i)) tokens[start].uriType = 1; // Raw based URI
+  else if (checkUri2(i)) tokens[start].uriType = 2; // Non-raw based URI
+  else return 0;
+
+  return leftParenthesis.right - start;
 }
 
 /**
- * Get node with URI
- * @returns {Array} `['uri', x]` where `x` is URI's nodes (without `url`
- *      and braces, e.g. `['string', ''/css/styles.css'']`).
+ * Get specific type of URI node
+ * @returns {Node|number} Specific type of URI node
  */
 function getUri() {
-  let startPos = pos;
-  let uriExcluding = {};
-  let uri;
-  let token;
+  const uriType = tokens[pos].uriType;
+
+  if (uriType === 1) return getUri1();
+  else if (uriType === 2) return getUri2();
+  else return 0;
+}
+
+/**
+ * Check if token type is valid URI character
+ * @param {number} i Token's index number
+ * @returns {number} Whether or not token is valid
+ */
+function _checkUriRawCharacters(i) {
+  switch (tokens[i].type) {
+    case TokenType.ExclamationMark:
+    case TokenType.QuotationMark:
+    case TokenType.NumberSign:
+    case TokenType.DollarSign:
+    case TokenType.PercentSign:
+    case TokenType.Ampersand:
+    case TokenType.Apostrophe:
+    case TokenType.Asterisk:
+    case TokenType.PlusSign:
+    case TokenType.Comma:
+    case TokenType.HyphenMinus:
+    case TokenType.FullStop:
+    case TokenType.Solidus:
+    case TokenType.Colon:
+    case TokenType.Semicolon:
+    case TokenType.EqualsSign:
+    case TokenType.QuestionMark:
+    case TokenType.CommercialAt:
+    case TokenType.LeftSquareBracket:
+    case TokenType.RightSquareBracket:
+    case TokenType.LowLine:
+    case TokenType.Tilde:
+      return 1;
+
+    default:
+      return 0;
+  }
+}
+
+/**
+ * Check if content of URI can be contained within a raw node
+ * @param {number} i Token's index number
+ * @returns {number} Length of raw node
+ */
+function checkUriRaw(i) {
+  const start = i;
   let l;
-  let raw;
 
-  pos += 2;
+  while (i < tokensLength) {
+    if (checkInterpolation(i)) break;
 
-  uriExcluding[TokenType.Space] = 1;
-  uriExcluding[TokenType.Tab] = 1;
-  uriExcluding[TokenType.Newline] = 1;
-  uriExcluding[TokenType.LeftParenthesis] = 1;
-  uriExcluding[TokenType.RightParenthesis] = 1;
-
-  if (checkUriContent(pos)) {
-    uri = []
-        .concat(getSC())
-        .concat(getUriContent())
-        .concat(getSC());
-  } else {
-    uri = [].concat(getSC());
-    l = checkExcluding(uriExcluding, pos);
-    token = tokens[pos];
-    raw = newNode(NodeType.RawType, joinValues(pos, pos + l), token.ln,
-        token.col);
+    if (l = checkIdent(i)) i += l;
+    else if (l = checkNumber(i)) i += l;
+    else if (l = _checkUriRawCharacters(i)) i += l;
+    // As comments are not allowed within the uri token, include as part of node
+    else if (l = checkCommentML(i)) i += l;
+    else break;
+  }
 
-    uri.push(raw);
+  return i - start;
```

Comment:
You can store position of final token as `uri_end`, for example, in which case `getUriRaw` will not need to loop through characters again to find it.